Source/DafnyCore/AST/TypeDeclarations/DatatypeCtor.cs

```diff
@@ -50,4 +50,8 @@ public DatatypeCtor(RangeToken rangeToken, Name name, bool isGhost, [Captured] L
   }
 
   public DafnySymbolKind Kind => DafnySymbolKind.EnumMember;
+  public string GetHoverText(DafnyOptions options) {
+    var formals = string.Join(", ", Formals.Select(f => f.AsText()));
```

Comment:
I saw this code snippet several times in this PR. Would it make sense to refactor it somewhere?

Reply:
It occurs twice, in DatatypeCtor and Function, but the only part that's exactly the same is `string.Join(", ", Formals.Select(f => f.AsText()))`. If there was multiple inheritance I might introduce something like `HasParameters` and include this method and the `Formals` field there, but given that there's not I'm more inclined to leave this like it is. Note there's also `string.Join(", ", Ins.Select(i => i.AsText()))` in `Method`.

I could introduce `interface IHasInputs { List<Formal> Inputs {get;} }` and have Function Method and DatatypeCtor inherit from it, and try to find all the existing duplications between these three, but that'd be better to do in a follow-up.